### PR TITLE
Fix 10438 - Remove parameters of private methods and constructors in declaration files

### LINKED ISSUE #10438: Private methods in d.ts files should be declared with rest parameters to make them more stable
**TypeScript Version:**  1.8.0 / nightly (2.0.0-dev.201xxxxx)

**Code**

```ts
export class Bug {
    private method(one: number, two: string): string {
    }
}

```

To make the d.ts more stable I think that the generated d.ts file should look like this:

```ts
export declare class Bug {
   private method(...params); 
}
```

Since the actual number of params of that method doesn't matter. All that is need is the name. This keeps the `shape` of a TS module more stable if private members change.

### PR DESCRIPTION
**Pull request by Igorbek**
Fixes #10438 

This PR makes these changes in declaration emitting:
- private instance methods to be emitted as `private methodName;`
- private static methods to be emitted as `private static methodName;`
- private constructor to be emitted as `private constructor();`
- if private method or constructor has overloads then only single declaration is emitted
- instance and static members with the same name are not considered overloads

Open questions:
- [x] is it ok to emit private constructors with removed parameters?
  - I think it is ok, since nobody would be able to call it anyway

_#myfirsttypescriptpr_

### CODE REVIEW

**Comment by Igorbek** [comment]
Is it going to be reviewed/merged?

File: src/compiler/declarationEmitter.ts [hunk 1]
```diff
@@ -1352,6 +1360,15 @@ namespace ts {
             // If we are emitting Method/Constructor it isn't moduleElement and hence already determined to be emitting
             // so no need to verify if the declaration is visible
             if (!resolver.isImplementationOfOverload(node)) {
+                // If we are emitting a private method or constructor it might be already emitted with an overloaded signature
```
Comment (sheetalkamat) [inline src/compiler/declarationEmitter.ts]: This seems incorrect way of handling it.
You could just check in emitSignatureDeclaration if this is first signature and skip rest for private methods.
Map of name is not needed and if needed you can ask resolver for the question instead.

Comment (sheetalkamat) [inline src/compiler/declarationEmitter.ts]: This seems incorrect way of handling it.
You could just check in emitSignatureDeclaration if this is first signature and skip rest for private methods.
Map of name is not needed and if needed you can ask resolver for the question instead.

Comment (sheetalkamat) [inline src/compiler/declarationEmitter.ts]: This seems incorrect way of handling it.
You could just check in ```emitSignatureDeclaration``` if this is first signature and skip rest for private methods.
Map of name is not needed and if needed you can ask resolver for the question instead.

Comment (Igorbek) [inline src/compiler/declarationEmitter.ts]: Thank you, I'll change

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has failing CI tests and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**